big packaging refactor

ctdk opened the pull request:
Adds:

ubuntu 14.04 with upstart
ubuntu 16.04, debian jessie with systemd
centos6 with upstart 0.6.5
centos7 with systemd

The existing sysvinit configuration was defined as debian/wheezy.

ctdk committed:
big packaging refactor

woodsaj commented:
lets move the NUMWORKERS to /etc/default/graphite-metrictank along with GRAPHITE_API_CONFIG and just make it static with a value of 2.

Then the systemd units should use `ENVIRONMENT_FILE=/etc/default/graphite-metrictank` and the init.d script can just source the file.

for upstart, you will need to something like

```
script
   set -a
   . /etc/default/graphite-metrictank
   exec chroot --userspec root:root / /usr/share/python/graphite/bin/python "/usr/share/python/graphite/bin/gunicorn" "graphite_api.app:app" "-b" "0.0.0.0:8888" "-k" "eventlet" "-w" "$NUMWORKERS" "--log-file" "/var/log/graphite/graphite-metrictank.log"
end script
```

woodsaj committed:
refactor packaging scripts

woodsaj committed:
update circleci config

woodsaj committed:
revert to using normal python instead of pypy

woodsaj committed:
add support for building docker images

woodsaj committed:
refactor build scripts.

Python is a pain to build for.  Every distro has different versions
of python and it was impossible to build binaries that were portable.
So to get working packages for all distros, we now just build the
binaries inside a docker container running the target distro/release.

woodsaj committed:
update cirlcleci config

woodsaj committed:
remove debug early exit

woodsaj committed:
bug fixes in packaging

woodsaj committed:
fix command to copy rootfs to builddir

woodsaj committed:
bug fix

woodsaj committed:
create build dir for target

woodsaj committed:
install python-pip into centos containers

woodsaj committed:
fix source path of graphite config

woodsaj committed:
fix path to post-install scripts

woodsaj committed:
create build dir on the host.

If the builddir is created by a build running in docker, it will
be owned by root

woodsaj committed:
update artifacts dir

woodsaj committed:
fix up copying common config files

woodsaj committed:
install rpm

woodsaj committed:
remove debug early exit

ctdk committed:
upstart config for centos 6 needs moved up a level

ctdk committed:
.gitignore was excluding the systemd service definitions (grrr)
